contracts/assemblyscript/nep4-basic/__tests__/main.unit.spec.ts

```diff
@@ -86,25 +87,77 @@ describe('transfer', () => {
 
     // Bob transfers to himself
     Context.setPredecessor_account_id(bob)
-    transfer(bob, aliceToken)
+    transfer_from(alice, bob, aliceToken)
 
     // it works!
     expect(get_token_owner(aliceToken)).toBe(bob)
     expect(get_token_owner(aliceToken)).not.toBe(alice)
   })
 
-  it('prevents anyone else from transferring the token', () => {
+  it('prevents escrow from using transfer_from given `token_id` to given `new_owner_id if `owner_id` does not match`', () => {
+    expect(() => {
+      // Alice grants access to Bob
+      Context.setPredecessor_account_id(alice)
+      grant_access(bob)
+
+      // Alice has a token
+      const aliceToken = nonSpec.mint_to(alice)
+      expect(get_token_owner(aliceToken)).toBe(alice)
+      expect(get_token_owner(aliceToken)).not.toBe(bob)
+
+      // Bob attempts to transfer and has access, but owner_id is wrong
+      Context.setPredecessor_account_id(bob)
+      transfer_from(bob, carol, aliceToken)
+    }).toThrow(nonSpec.ERROR_OWNER_ID_DOES_NOT_MATCH_EXPECTATION)
+  })
+
+  it('prevents anyone else from using transfer_from on the token', () => {
     expect(() => {
       // Alice has a token
       const aliceToken = nonSpec.mint_to(alice)
 
       // Bob tries to transfer it to himself
       Context.setPredecessor_account_id(bob)
-      transfer(bob, aliceToken)
+      transfer_from(alice, bob, aliceToken)
     }).toThrow(nonSpec.ERROR_CALLER_ID_DOES_NOT_MATCH_EXPECTATION)
   })
 })
 
+describe('transfer', () => {
+  it('allows owner to transfer given `token_id` to given `owner_id`', () => {
+    // Alice has a token
+    const aliceToken = nonSpec.mint_to(alice)
+    expect(get_token_owner(aliceToken)).toBe(alice)
+    expect(get_token_owner(aliceToken)).not.toBe(bob)
+
+    // Alice transfers her token to Bob
+    Context.setPredecessor_account_id(alice)
+    transfer(bob, aliceToken)
+
+    // it works!
+    expect(get_token_owner(aliceToken)).toBe(bob)
+    expect(get_token_owner(aliceToken)).not.toBe(alice)
+  })
+  
+  it('prevents escrow from using transfer. Escrow can only use transfer_from', () => {
```

Comment:
I'm torn on whether it makes sense to add tests for "everyone else" in addition to escrow, as we do above for `transfer_from`. What was your thought process on not adding it?